src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2018, Ford Motor Company
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include "rc_rpc_plugin/interior_data_cache_impl.h"
+#include "utils/date_time.h"
+#include "utils/logger.h"
+
+namespace rc_rpc_plugin {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule");
+
+InteriorDataCacheImpl::InteriorDataCacheImpl(
+    const uint32_t time_frame_of_allowed_requests)
+    : reset_request_count_timer_(
+          "InteriorDataCacheImplResetRequests",
+          new timer::TimerTaskImpl<InteriorDataCacheImpl>(
+              this, &InteriorDataCacheImpl::ResetRequestCountOnTimer)) {
+  StartRequestResetTimer(time_frame_of_allowed_requests);
```

Comment:
@LuxoftAKutsan why we start timer here?

Reply:
@LitvinenkoIra @yang1070 this is the periodical timer that should just reset counter once a time period, starting timer on first interior data will make logic much more complicated. We should somehow check that it is first getinteriorvehicleData, and stop if there were no interior vehicle data requests for some time period. I see the solution with periodical timer more lightweight

Reply:
@LitvinenkoIra @yang1070  

Expected behaviour of SDL: 
On each GIVR that SDL want to send to HMI, SDL should get history of last 5 (first parameter of ini file) requests by module type and look if the oldest one was more than 1 (second parameter of ini file) seconds ago to prevent out of bounce amount of requests to HMI in the borders of the time window.